### DISCUSSION TITLE
Escaping characters when calling shell commands

### DISCUSSION THREAD

**Discussion by ristomatti** [houmain/keymapper#165] [Q&A] [answered]
I tried adding an alias to run a command what takes a hex color value as an argument, e.g.:

```bash
trigger = $(pastel --force-color paint "#cb4b16" "[trigger = $0]" >> LOG_FILE)
```

This fails with `ERROR: Unterminated string in line x` as `#` gets interpreted as the beginning of a comment. I tried escaping it using the common `\` and `\\` syntax but it still results in the same error.

Is there some undocumented syntax to do this by any chance? I could've reported this as a feature request, but it's just a minor inconvenience. In this particular case it turned out `pastel` doesn't require prefixing the value `#`.

Similarly it's not possible to escape characters `[]` in macro arguments (noticed this yesterday).

I probably should start writing shell script wrappers to this stuff anyhow to make things cleaner. :grin:

**Comment by houmain** [comment, marked as answer]
Thanks for the report!
No, there is no way to escape these. # and ] in strings should not be interpreted.
I think this is worth an `issue` but it does not really matter, both will be fixed soon.

**Reply by ristomatti** [reply]
Issue gone after installing 4.6.0. :rocket:
